### ISSUE TITLE
Unable to log into virtual studio account on Linux

### ISSUE THREAD

**Issue by mikedickey** [jacktrip/jacktrip#561]
Using the most recent builds, and I tried both Flatpak and static binary. On Fedora 35.

If I try to login, it opens a browser window that says:

```
Callback URL mismatch.
The provided redirect_uri is not in the list of allowed callback URLs.
```

Looking at the URL, it has `redirect_uri=http://127.0.0.1:0/` (0 is not a valid port number)

**Comment by mikedickey** [comment]
```➜  ~ jacktrip
Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.
SSL Library: OpenSSL 1.1.1n  FIPS 15 Mar 2022
Opening in existing browser session.
libva error: vaGetDriverNameByIndex() failed with unknown libva error, driver_name = (null)
[187749:187749:0100/000000.068282:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.```

**Comment by mikedickey** [comment]
Turns out something else was using port 42424. The app needs this to be available when you try to log into virtual studio; otherwise, it fails.

**Comment by psiborg112** [comment]
I'm a bit swamped at the moment, but should be able to get to this sometime next week. The big question is how forgiving Auth0 is on us using at least a range of port numbers.

**Comment by psiborg112** [comment]
If Auth0 will allow us to use any port, then it's a really simple fix. If not, I should be able to check if the TCP server that it creates is successfully listening on the required port, and if not, fall back to another port in a range that we decide on.

**Comment by ntonnaett** [comment]
If this is TCP we should better use ephemeral ports.

**Comment by psiborg112** [comment]
Agreed

**Comment by ntonnaett** [comment]
@psiborg112 Like so: #562 ?

**Comment by psiborg112** [comment]
Unfortunately it's not that simple. If it were, the easier way to do it would be to change:
`m_authenticator->setReplyHandler(new QOAuthHttpServerReplyHandler(port, this));`
to 
`m_authenticator->setReplyHandler(new QOAuthHttpServerReplyHandler(this));`
a few lines later.

Instead, it looks like we have to choose a specific port range to use (with a go to port to test first) because Auth0 doesn't allow a wildcard port in the list of authorised callback URLs. This needs to done in conjunction with the virtual studio folk because they need to change the Auth0 settings.

**Comment by OConnorStP** [comment]
I am always happy to try images on Debian 11.  Can somebody hold my hand and direct me to where they are?

**Comment by mattahorton** [comment]
I made #563 to address this. We can't use ephemeral ports due to Auth0's limitations.

**Comment by psiborg112** [comment]
We can't use an auto assigned ports, but ephemeral ports are just anything in the 49152–65535 range based on the IANA definition of them. So #563 does exactly what we need it to do. It might still be worth having some code at some stage that sends us to a fallback port, but this should make the need for that less likely.

**Comment by psiborg112** [comment]
I'm going to close this since I've just merged #563.
